### PR TITLE
Release 1.2.0

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Review by ekohl — COMMENTED** [review]
Semantically I'd expect 1.2.0 since I'd consider it more of an enhancement than a bugfix.

**Comment by ehelms** [comment]
The reason I considered it a bug fix is the following:

```
fatal: [pipe-katello-proxy-nightly-centos7]: FAILED! => changed=true 
  cmd: |-
    foreman-installer -v --no-colors --disable-system-checks --scenario foreman-proxy-content  --certs-tar-file /root/pipe-katello-proxy-nightly-centos7.wareagle.example.com.tar.gz --foreman-proxy-trusted-hosts pipe-katello-server-nightly-centos7.wareagle.example.com --foreman-proxy-trusted-hosts pipe-katello-proxy-nightly-centos7.wareagle.example.com --foreman-proxy-foreman-base-url https://pipe-katello-server-nightly-centos7.wareagle.example.com --foreman-proxy-oauth-consumer-key TertzjwwEQSLtzpAobxoHDe3CR8CncZY --foreman-proxy-oauth-consumer-secret ysxfUb4pn9xZHkGQpgQTKdtxNBmC3EVR  --puppet-server-foreman-url https://pipe-katello-server-nightly-centos7.wareagle.example.com --puppet-server-max-active-instances 1 --puppet-server-jvm-min-heap-size 1G --puppet-server-jvm-max-heap-size 1G
  delta: '0:00:02.633433'
  end: '2021-03-16 01:37:58.977077'
  msg: non-zero return code
  rc: 1
  start: '2021-03-16 01:37:56.343644'
  stderr: |-
    /opt/rh/rh-ruby27/root/usr/share/ruby/open3.rb:213:in `spawn': No such file or directory - puppet (Errno::ENOENT)
            from /opt/rh/rh-ruby27/root/usr/share/ruby/open3.rb:213:in `popen_run'
            from /opt/rh/rh-ruby27/root/usr/share/ruby/open3.rb:101:in `popen3'
            from /opt/rh/rh-ruby27/root/usr/share/ruby/open3.rb:281:in `capture3'
            from /opt/theforeman/tfm/root/usr/share/gems/gems/kafo_parsers-1.1.1/lib/kafo_parsers/puppet_strings_module_parser.rb:141:in `run_puppet'
            from /opt/theforeman/tfm/root/usr/share/gems/gems/kafo_parsers-1.1.1/lib/kafo_parsers/puppet_strings_module_parser.rb:30:in `available?'
```

This actually breaks on Ruby 2.7 via Ansible running it.

**Comment by ekohl** [comment]
AFAIK `kafo_parsers` only executes Puppet if there is no (up to date) parser cache. That only happens when you don't have a cache or if you modified Puppet files since the cache was generated.

**Comment by ehelms** [comment]
> AFAIK `kafo_parsers` only executes Puppet if there is no (up to date) parser cache. That only happens when you don't have a cache or if you modified Puppet files since the cache was generated.

Strange, as I did not modify anything. If I run the installer directly, this does not happen. If I run Ansible scripts, this does happen. And only when I have been testing Ruby 2.7 packages. Note that it does not actually call to create a parser cache, it appears to only get so far as checking which parser is available. 

```
[root@pipe-katello-server-nightly-centos7 vagrant]# foreman-installer
FIND PUPPET BIN
/opt/rh/rh-ruby27/root/usr/local/bin
false
/opt/rh/rh-ruby27/root/usr/bin
false
/opt/theforeman/tfm/root/usr/bin
false
/sbin
false
/bin
false
/usr/sbin
false
/usr/bin
false
/opt/puppetlabs/bin
false
/opt/puppetlabs/bin
false
FOUND PUPPET PATH: 
ENV
puppet
puppet
help
strings
2021-03-16 01:18:49 [NOTICE] [root] Loading default values from puppet modules...
2021-03-16 01:18:52 [NOTICE] [root] ... finished
```

This is I believe due to when loading default module values, it initializes the PuppetModule class in Kafo which triggers `find_parser`: https://github.com/theforeman/kafo/blob/master/lib/kafo/puppet_module.rb#L28

**Comment by ekohl** [comment]
I would argue that that's kafo's problem. Semantically as an API user you wouldn't expect it to change the behavior like this in a patch version.

**Comment by ehelms** [comment]
I don't understand what behavior is changing. But, I'll update to a 1.2.0.

File: lib/kafo_parsers/version.rb [hunk 1]
```diff
@@ -1,3 +1,3 @@
 module KafoParsers
-  VERSION = "1.1.2"
+  VERSION = "1.1.3"
```
Comment (ekohl) [inline lib/kafo_parsers/version.rb:2]: You missed the actual code change :)

There is actually a config file:
https://github.com/theforeman/kafo_parsers/blob/master/.bumpversion.cfg

So you can run `bumpversion minor` and it'll do the right thing for a minor release. Including tagging if you want.

Comment (ehelms) [reply lib/kafo_parsers/version.rb]: I missed committing it. That code change was sitting there waiting for me to come back and find it.

**Review by ekohl — CHANGES_REQUESTED** [review]
Would you mind updating this too:
https://github.com/theforeman/kafo_parsers/blob/ffdb47eb2d6c85f6981700851d15643c45acdbf2/.bumpversion.cfg#L2

And next time use bumpversion (it's in Fedora) :)

**Review by ekohl — APPROVED** [review]

**Comment by ehelms** [comment]
> And next time use bumpversion (it's in Fedora) :)

Next time, I'll try. But given its not in our docs, its easy to miss. And updating the version.rb file and opening a PR is really quick and easy compared to going and remembering how to use bumpversion to do the same.

**Pull request merged**